### PR TITLE
Command HA with multiple config

### PR DESCRIPTION
**Pull request by shifali19**

### CODE REVIEW

**Comment by shifali19** [comment]
recheck ha

**Comment by shifali19** [comment]
recheck ha

**Comment by shifali19** [comment]
recheck ha

**Review by mfigurski — CHANGES_REQUESTED** [review]

File: pkg/controller/command/command_controller.go [hunk 1]
```diff
@@ -567,3 +625,81 @@ func (r *ReconcileCommand) getConfig(command *contrail.Command) (*contrail.Confi
...
+	}, webUI)
+
+	return webUI, err
+}
+
+func (r *ReconcileCommand) reconcileBootstrapJob(command *contrail.Command, commandBootStrapConfigName string) error {
+	bootstrapJob := &batch.Job{}
+	jobName := types.NamespacedName{Namespace: command.Namespace, Name: command.Name + "-bootstrap-job"}
+	err := r.client.Get(context.Background(), jobName, bootstrapJob)
+	alreadyExists := err == nil
```
Comment (kaweue) [inline pkg/controller/command/command_controller.go:643]: Shouldn't we check if error is not found?
In my opinion we should return err if error is different then in not found

File: pkg/controller/command/command_controller.go [hunk 2]
```diff
@@ -372,6 +429,14 @@ func (r *ReconcileCommand) getKeystone(command *contrail.Command) (*contrail.Key
 }
 
 func newDeployment(name, namespace, configVolumeName string, csrSignerCaVolumeName string, containers []*contrail.Container) *apps.Deployment {
+	podIPEnv := core.EnvVar{
+		Name: "MY_POD_IP",
```
Comment (kaweue) [inline pkg/controller/command/command_controller.go:433]: Maybe "POD_IP"

**Review by kaweue — CHANGES_REQUESTED** [review]

File: pkg/controller/command/command_controller.go [hunk 3]
```diff
@@ -424,6 +489,9 @@ func newDeployment(name, namespace, configVolumeName string, csrSignerCaVolumeNa
 								MountPath: certificates.SignerCAMountPath,
 							},
 						},
+						Env: []core.EnvVar{
```
[the remark this answers is not in the document]
Comment (shifali19) [reply pkg/controller/command/command_controller.go]: Not applicable anymore.

File: pkg/controller/command/command_controller.go [hunk 4]
```diff
@@ -476,7 +534,7 @@ func getImage(containers []*contrail.Container, containerName string) string {
 func getCommand(containers []*contrail.Container, containerName string) []string {
 	var defaultContainersCommand = map[string][]string{
 		"init":                {"bash", "-c", "/etc/contrail/bootstrap.sh"},
-		"api":                 {"bash", "-c", "/etc/contrail/entrypoint.sh"},
+		"api":                 {"bash", "-c", "/etc/contrail/entrypoint${MY_POD_IP}.sh"},
```
[the remark this answers is not in the document]
Comment (shifali19) [reply pkg/controller/command/command_controller.go]: Agreed.Done

File: pkg/controller/command/command_controller.go (hunk 1 above)
Comment (shifali19) [reply pkg/controller/command/command_controller.go:631]: Done

File: pkg/controller/command/command_controller.go (hunk 2 above)
Comment (shifali19) [reply pkg/controller/command/command_controller.go]: Not applicable anymore.

**Review by kaweue — APPROVED** [review]

**Review by mfigurski — APPROVED** [review]

**Pull request merged**